Analytics/resources/sendgrid_management.py

```diff
@@ -0,0 +1,195 @@
+        while line:
+            if "export SENDGRID_API_KEY=" in line:
+                found = True
+                line = "export SENDGRID_API_KEY='{}'".format(new_api_key)
+
+            new_lines.append(line)
+            line = filein.readline()
+
+        new_lines.append('\n')
+        filein.close()
+
+        if found:
+            fileout = open(file_path, 'w')
+            fileout.writelines(new_lines)
+            fileout.close()
+            return True, None
+        else:
+            logger.error("The required `export SENDGRID_API_KEY` statement "
+                         "is not present within{}".format(file_path))
+            return False, "Could not replace API key as it does not exist"
+
+
+class SendgridHelper:
+    """ Helper Class. Contains methods used by Sendgrid Resource classes """
+
+    @staticmethod
+    def send_test_request(sendgrid_api_key: str) -> requests.Response:
+        """
+        Send POST request to the Sendgrid test endpoint containing
+        the the supplied API key
+        :param sendgrid_api_key: Key whose validity is to be tested
+        :return: The HTTP response received from Sendgrid
+        """
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": "Bearer {}".format(sendgrid_api_key)
+        }
+        data = {"personalizations":
+                    [{"to": [{"email": "recipient@example.com"}]}],
+                "from": {"email": "sendeexampexample@example.com"},
+                "subject": "HelloWorld!",
+                "content": [{"type": "text/plain", "value": "Howdy!"}]
+                }
+
+        return requests.post("https://api.sendgrid.com/v3/mail/send",
+                             headers=headers, json=data)
+
+    @staticmethod
+    def handle_sendgrid_response(
+            sendgrid_response: requests.Response) -> (dict, int):
+        """
+        Return the applicable feedback according to the supplied Sendgrid
+        response
+        :param sendgrid_response: An HTTP response from the Sengrid test
+                                  endpoint
+        :return: a dictionary containing whether the API key is valid and
+                 the corresponding response code
+        """
+
+        if sendgrid_response.status_code == 202:
+            return {"api_key": True, "reason": ""}, 200
+        elif sendgrid_response.status_code == 401:
+            logger.critical("Current Sendgrid API key environment variable is "
+                            "invalid. Please replace this environment "
+                            "variable with a new authorized Sengrid API key "
+                            "and replace the export statement in "
+                            "~/.bash_profile with this key")
+            response_json = sendgrid_response.json()
+            return {
+                       "api_key": False,
+                       "reason": response_json["errors"][0]["message"]
+                   }, 200
```

Comment:
Is this correct to return 200 when the reason is response_json["errors"][0]["message"]?

Reply:
The 200 is indicate that process executed correctly. The response_json["errors"][0]["message"] is the error message from Sendgrid that indicates why the api key is not valid